### PR TITLE
TST: move and reformat latex_na_rep test

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Move test to a more suitable location (``class TestToLatexFormatters``)
and reformat using ``_dedent`` and raw string for better readability.

### CODE REVIEW

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: pandas/tests/io/formats/test_to_latex.py [hunk 1]
```diff
@@ -972,6 +972,30 @@ def test_to_latex_float_format_no_fixed_width_integer(self):
...
+    def test_to_latex_na_rep_and_float_format(self, na_rep):
+        df = DataFrame(
+            [
+                ["A", 1.2225],
+                ["A", None],
+            ],
+            columns=["Group", "Data"],
+        )
+        result = df.to_latex(na_rep=na_rep, float_format="{:.2f}".format)
+        expected = _dedent(
```
Comment (dsaxton) [inline pandas/tests/io/formats/test_to_latex.py:985]: Nice. I think this same pattern appears in other places, could push this definition higher up and use elsewhere?

**Review by dsaxton — APPROVED** [review]

File: pandas/tests/io/formats/test_to_latex.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/tests/io/formats/test_to_latex.py:985]: Do you mean ``_dedent`` (without the preceding newline?)

Comment (dsaxton) [reply pandas/tests/io/formats/test_to_latex.py:985]: Ah, maybe they aren't using the preceding newline so could probably use dedent alone for those

Comment (ivanovmg) [reply pandas/tests/io/formats/test_to_latex.py:985]: Here I used custom ``_dedent`` only to make it compatible with raw strings.
In general I would simply use this
```
dedent(
    """\
    content
    """
)
```

**Comment by jreback** [comment]
thanks @ivanovmg

**Pull request merged**